File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.cpp
// Last updated: 10/08/2026, 02:36:13
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;  // handle empty case
        
        int n = nums.size();
        int count = 1;
        int maxCount = 1;
        sort(nums.begin(), nums.end());
        
        for (int i = 0; i < n - 1; i++) {
            // If the next element is a duplicate, continue without resetting or increasing the count
            if (nums[i] == nums[i+1]) continue;
            
            if (nums[i] + 1 == nums[i+1]) {
                count++;
                maxCount = max(maxCount, count);
            } else {
                count = 1;  // Reset to 1 because the next number is the start of a new sequence
            }
        }
        return maxCount;
    }
};

File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.py
# Last updated: 10/08/2026, 02:36:08
class Solution(object):
    def longestConsecutive(self, nums):
        num_set = set(nums)
        best = 0
        for num in num_set:
            if num - 1 not in num_set:
                length = 1
                while num + length in num_set:
                    length += 1
                best = max(best, length)
        return best
        